File: api/health.js
const { getDbPool, getDbProviderInfo } = require('./_lib/db');
const { handleOptions, setCorsHeaders } = require('./_lib/cors');

module.exports = async (req, res) => {
  if (handleOptions(req, res, 'GET, OPTIONS')) return;
  setCorsHeaders(res, 'GET, OPTIONS');

  const pool = getDbPool();
  const providerInfo = getDbProviderInfo();
  let dbStatus = pool ? "INITIALIZING" : (providerInfo.status === "FROZEN_BLOCKED" ? "NEON_FROZEN_AIVEN_REQUIRED" : "NOT_CONFIGURED");
  let counts = {};
  let pgVersion = null;

  if (pool) {
    try {
      const [c1, c2, c3, vRes] = await Promise.all([
        pool.query("SELECT COUNT(*) FROM verified_factchecks;"),
        pool.query("SELECT COUNT(*) FROM raw_trends_inbox;"),
        pool.query("SELECT COUNT(*) FROM raw_trends_inbox WHERE triage_status = 'QUEUED_FOR_INVESTIGATION';"),
        pool.query("SELECT version();")
      ]);
      counts["verified_factchecks"] = parseInt(c1.rows[0].count, 10);
      counts["raw_trends_inbox"] = parseInt(c2.rows[0].count, 10);
      counts["queued_for_investigation"] = parseInt(c3.rows[0].count, 10);
      pgVersion = vRes.rows[0]?.version || '';
      dbStatus = "CONNECTED_HEALTHY";
    } catch (e) {
      console.error('[Health Check DB Query Error]:', e.message);
      dbStatus = "UNHEALTHY";
    }
  }

  const responsePayload = {
    service: "AI Tech-Lineage Fact-Check Hub (Vercel Serverless Node.js Backend)",
    version: "v21.2",
    database_status: dbStatus,
    database_provider: providerInfo.provider,
    database_host: providerInfo.host,
    database_version: pgVersion,
    neon_postgres_status: providerInfo.provider === 'Aiven PostgreSQL' ? 'DECOMMISSIONED_IN_FAVOR_OF_AIVEN' : (providerInfo.status === 'FROZEN_BLOCKED' ? 'BLOCKED_FROZEN' : dbStatus),
    database_url_present: Boolean(pool),
    metrics: counts
  };

  if (providerInfo.instruction) {
    responsePayload.instruction = providerInfo.instruction;
  }

  return res.status(200).json(responsePayload);
};

File: api/health.py
from http.server import BaseHTTPRequestHandler
import json
import os
import psycopg2

NEON_DATABASE_URL = os.environ.get(
    "DATABASE_URL",
    "postgresql://neondb_owner:npg_1sXv4iFvJadO@ep-quiet-grass-a1wqqs32-pooler.ap-southeast-1.aws.neon.tech/neondb?sslmode=require"
)

class handler(BaseHTTPRequestHandler):
    def do_GET(self):
        db_status = "UNKNOWN"
        counts = {}
        try:
            conn = psycopg2.connect(NEON_DATABASE_URL)
            cur = conn.cursor()
            cur.execute("SELECT COUNT(*) FROM verified_portfolios;")
            counts["verified_portfolios"] = cur.fetchone()[0]
            cur.execute("SELECT COUNT(*) FROM inbox_candidates;")
            counts["inbox_candidates"] = cur.fetchone()[0]
            cur.execute("SELECT COUNT(*) FROM inbox_candidates WHERE status = 'QUEUED_FOR_INVESTIGATION';")
            counts["queued_for_investigation"] = cur.fetchone()[0]
            cur.close()
            conn.close()
            db_status = "CONNECTED_HEALTHY"
        except Exception as e:
            db_status = f"ERROR: {str(e)}"

        self.send_response(200)
        self.send_header('Content-Type', 'application/json; charset=utf-8')
        self.send_header('Access-Control-Allow-Origin', '*')
        self.end_headers()

        res = {
            "service": "AI Tech-Lineage Fact-Check Hub (Vercel Serverless Backend)",
            "version": "v14.0",
            "neon_postgres_status": db_status,
            "metrics": counts
        }
        self.wfile.write(json.dumps(res, ensure_ascii=False, indent=2).encode('utf-8'))
